worker/oci/spec_unix.go

```diff
@@ -18,8 +20,17 @@ import (
 
 // Ideally we don't have to import whole containerd just for the default spec
 
-func GenerateSpec(ctx context.Context, meta worker.Meta, mounts []worker.Mount) (*specs.Spec, func(), error) {
-	s, err := containerd.GenerateSpec(ctx, nil, nil,
+// GenerateSpec generates spec using containerd functionality.
+func GenerateSpec(ctx context.Context, meta worker.Meta, mounts []worker.Mount, id string) (*specs.Spec, func(), error) {
+	c := &containers.Container{
+		ID: id,
+	}
+	_, ok := namespaces.Namespace(ctx)
```

Comment:
Can you add a comment about why this is needed? Possibly something that may need further discussion in `containerd` repo. Very confusing why a `GenerateSpec()` function requires a container and client parameter.